lib/spack/docs/configuration.rst

```diff
@@ -261,3 +261,52 @@ The merged configuration would look like this:
        - /lustre-scratch/$user
        - ~/mystage
    $ _
+
+------------------------------
+Config file variables
+------------------------------
+
+Spack understands several variables which can be used in config file paths
+where ever they appear. There are three sets of these variables, Special 
```

Comment:
I think a name should be chosen to indicate these variables are understood by Spack. "special" is too general. My suggestion is "spack-specific variables" but anything more specific is great.

Also nitpick: the order presented here is special, environment, and user variables. But the order of presentation is special, user, environment. I'd prefer the order of the sections matches the order here (in part because user variables differ from environment and special variables).